library/core/src/iter/traits/iterator.rs

```diff
@@ -69,7 +69,7 @@ fn _assert_is_object_safe(_: &dyn Iterator<Item = ()>) {}
     message = "`{Self}` is not an iterator"
 )]
 #[doc(notable_trait)]
-#[rustc_diagnostic_item = "Iterator"]
+#[cfg_attr(not(bootstrap), lang = "iterator")]
```

Comment:
we could also keep the diagnostic item along with the lang item, which would avoid the diff of almost this entire commit